src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -452,3 +471,33 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         )
         df._reader = self
         return df
+
+    def _upload_local_file_to_stage(self, path: str, format: str) -> str:
+        ostype = os.name
+        temp_stage = self._session.get_session_stage()
+        if os.path.exists(path) and os.path.isfile(path):
+            _ = self._session.file.put(path, temp_stage, auto_compress=False)
+            _, filename = os.path.split(path)
+            path = os.path.join(temp_stage, filename)
+            if ostype == "nt":
+                path = path.replace("/", "\\")
+            else:
+                path = path.replace("\\", "/")
+        if os.path.exists(path) and os.path.isdir(path):
+            if format != "csv":
+                raise ValueError(
+                    f"Only support a local directory for reading CSV files, but got {format}"
+                )
+            filelist = os.listdir(path)
+            for file in filelist:
+                if os.path.splitext(file)[-1][1:].lower() == format.lower():
+                    filepath = os.path.join(path, file)
+                    if ostype == "nt":
+                        filepath = filepath.replace("/", "\\")
+                    else:
+                        filepath = filepath.replace("\\", "/")
+                    _ = self._session.file.put(
```

Comment:
ditto

Reply:
already deleted, this is just trying to fix a bug before